File: w7/log_probability.py
#!/usr/bin/python
import sys,re,glob,math
words=[]
name=[]
for file in glob.glob("poems/*.txt"):
	total=0
	count=0

	for line in open(file):
        	words=re.findall(r'(\b[A-Za-z]+\b)',line)
        	for each in words:
			total+=1
			if(each.lower() == sys.argv[1].lower()):
				count+=1;
	result=math.log(float(count+1)/total)

	file = re.sub(r'_', ' ',file)
	file = re.sub('.txt','',file)
	file = re.sub('poems\/','',file)
	print "log((%d+1)/%6d) = %8.4f %s"%(count,total,result,file)

File: w7/log_probability.pl
#!/usr/bin/perl -w
foreach $file (glob "poems/*.txt") {
   open(F1,"$file") or die "$0: Can't open $file $!\n";
   open(F2,"$file") or die "$0: Can't open $file $!\n";
   $file =~ s/_/ /g;
   $file =~ s/.txt//g;
   $file =~ s/poems\///g;
   $counter = 0;
   @total = ();
   while ($line = <F1>) {
      @words = split /\W+/, $line;
      $total[$counter] = @words;
      if ($total[$counter] != 0) {
         $num = 0;
         while ($num < $total[$counter]) {
        	   if ($words[$num] !~ /[a-zA-Z]/) {
        		     $total[$counter] = $total[$counter] - 1;
        		}
        		$num++;
        	}
      }
      $counter ++;
   }
   $pos = 0;
   $sum = 0;
   while ($pos < $counter) {
   	$sum += $total[$pos];
   	$pos ++;
   }
   $all = 0;
   while ($line = <F2>) {
   	@words = split /\W+/, $line;
      $total = @words;
   	$num = 0;
   	while ($num < $total) {
   		if ($words[$num] =~/^$ARGV[0]$/i) {
            $all += 1;
   		}
   		$num++;
   	}
   }
   $loga = log(($all+1)/$sum);
   printf "log((%d+1)/%6d) = %8.4f %s\n", $all, $sum, $loga, $file;
   close(F1);
   close(F2);
}
